src/evo/deterministicmns.cpp

```diff
@@ -49,17 +49,13 @@ UniValue CDeterministicMN::ToJson() const
     obj.pushKV("collateralHash", collateralOutpoint.hash.ToString());
     obj.pushKV("collateralIndex", (int)collateralOutpoint.n);
 
-    CScript scriptPubKey;
-    if (Coin coin; GetUTXOCoin(collateralOutpoint, coin)) {
-        scriptPubKey = coin.out.scriptPubKey;
-    } else {
-        uint256 tmpHashBlock;
-        CTransactionRef collateralTx = GetTransaction(/* block_index */ nullptr,  /* mempool */ nullptr, collateralOutpoint.hash, Params().GetConsensus(), tmpHashBlock);
-        scriptPubKey = collateralTx->vout[collateralOutpoint.n].scriptPubKey;
-    }
-    CTxDestination dest;
-    if (ExtractDestination(scriptPubKey, dest)) {
-        obj.pushKV("collateralAddress", EncodeDestination(dest));
+    uint256 tmpHashBlock;
+    CTransactionRef collateralTx = GetTransaction(/* block_index */ nullptr,  /* mempool */ nullptr, collateralOutpoint.hash, Params().GetConsensus(), tmpHashBlock);
+    if (collateralTx) {
+        CTxDestination dest;
+        if (ExtractDestination(collateralTx->vout[collateralOutpoint.n].scriptPubKey, dest)) {
+            obj.pushKV("collateralAddress", EncodeDestination(dest));
+        }
```

Comment:
needs release notes documenting breaking change when txindex is disabled

Reply:
The following RPC calls are affected:

* `protx info` ([source](https://github.com/dashpay/dash/blob/4a520991db12ff1276aac0ea43a77fe9d400d52d/src/rpc/evo.cpp#L1469))
* `protx list` ([source](https://github.com/dashpay/dash/blob/4a520991db12ff1276aac0ea43a77fe9d400d52d/src/rpc/evo.cpp#L1376), [source](https://github.com/dashpay/dash/blob/4a520991db12ff1276aac0ea43a77fe9d400d52d/src/rpc/evo.cpp#L1402))
* `protx listdiff` ([source](https://github.com/dashpay/dash/blob/4a520991db12ff1276aac0ea43a77fe9d400d52d/src/rpc/evo.cpp#L1581-L1583))

Resolved in latest push.